### PR TITLE
Extend TF Import Support for Pool layers

### PR DESCRIPTION
**Pull request by Ram81**
@deshraj @utsavgarg In this PR I have extended support for Max Pooling 3D, Average Pooling 3D layer and added tests for the same. This PR depends on #352 , once it is merged there will be some minor changes in tests that need to be made.

### CODE REVIEW

File: tensorflow_app/views/import_graphdef.py [hunk 1]
```diff
@@ -55,17 +56,23 @@ def get_padding(node, layer):
     pad_top = pad_along_height / 2
     pad_left = pad_along_width / 2
     '''
-    if node.type == "Conv3D":
+
+    if node.type == "Conv3D" or node.type == "MaxPool3D" or node.type == "AvgPool3D":
```
Comment (kdexd) [inline tensorflow_app/views/import_graphdef.py]: `node.type in {"Conv3D", "MaxPool3D", "AvgPool3D"}` will make it cleaner and you can extend it without too much or conditions whenever you need to add more support of 3D layers.

File: tensorflow_app/views/import_graphdef.py [hunk 2]
```diff
@@ -244,8 +252,30 @@ def import_graph_def(request):
                         return JsonResponse({'result': 'error', 'error':
                                              'Missing shape info in GraphDef'})
 
+                if str(node.type) == 'MaxPool3D':
```
Comment (kdexd) [inline tensorflow_app/views/import_graphdef.py]: Why do you need to explicitly apply `str` method? I saw it in a few places above -- `node.type` is already a string. I am a bit confused myself.

File: tensorflow_app/views/import_graphdef.py [hunk 3]
```diff
@@ -262,6 +292,28 @@ def import_graph_def(request):
...
+                    layer['params']['stride_w'] = int(
+                        node.get_attr('strides')[3])
+                    layer['params']['layer_type'] = '3D'
+                    try:
+                        layer['params']['pad_d'], layer['params']['pad_h'], \
+                            layer['params']['pad_w'] = get_padding(node, layer)
+                    except TypeError:
+                        return JsonResponse({'result': 'error', 'error':
+                                             'Missing shape info in GraphDef'})
+
```
Comment (kdexd) [inline tensorflow_app/views/import_graphdef.py]: These lines are duplicated in `MaxPool3D`. You can try to make it hierarchical -- first, check if `node.type` has a substring`3D` and then check if it is max or average. That should reduce a lot of duplication.

Comment (kdexd) [reply tensorflow_app/views/import_graphdef.py]: Similar comment goes for `MaxPool` and `AvgPool` variants.

**Comment by kdexd** [comment]
How about aliasing `MaxPool` as `MaxPool2D` and `AvgPool` as `AvgPool2D`, I think Tensorflow layers are named like this too. It would avoid confusion.

Thoughts @Ram81 @deshraj @utsavgarg ?

**Comment by Ram81** [comment]
@kdexd the name's of ops in pbtxt are MaxPool and AvgPool that's why we use it. I am not sure if I get where you want aliasing.

**Comment by kdexd** [comment]
@Ram81 Oh okay, understood. Nevermind.

**Pull request merged**